WorkerLocationGenerator.html

```diff
@@ -0,0 +1,257 @@
+				Math.floor((worldCoordinate.y - worldCoordinateNW.y) * scale)
+			);
+			return currentLatLngOffset;
+		}
+		function setMenuXY(currentLatLng){
+			var mapWidth = $('#map_canvas').width();
+			var mapHeight = $('#map_canvas').height();
+			var menuWidth = $('.cmenu').width();
+			var menuHeight = $('.cmenu').height();
+			var clickedPosition = getCanvasXY(currentLatLng);
+			var x = clickedPosition.x ;
+			var y = clickedPosition.y ;
+
+			if((mapWidth - x ) < menuWidth)
+				 x = x - menuWidth;
+			if((mapHeight - y ) < menuHeight)
+				y = y - menuHeight;
+
+			$('.cmenu').css('left',x);
+			$('.cmenu').css('top',y);
+		};
+		function showContextMenu(currentLatLng) {
+			var projection;
+			var contextmenuDir;
+			projection = map.getProjection() ;
+			$('.cmenu').remove();
+			contextmenuDir = document.createElement("div");
+			contextmenuDir.className  = 'cmenu';
+			contextmenuDir.innerHTML = '<div class="item" title="Add another location to check" onclick="addNewArea' + currentLatLng + ';"><div class="label">Add Area</div></div><div class="item ' + (locations.length > 0 ? '' : 'disabled') + '" title="Copy locations list to clipboard" ' + (locations.length > 0 ? 'onclick="copyWorkerLocations()"' : '') + '><div class="label">Copy Marker Locations</div></div><div class="item ' + (locations.length > 0 ? '' : 'disabled') + '" title="Clear all areas from the map" ' + (locations.length > 0 ? 'onclick="clearAllAreas()"' : '') + '><div class="label">Clear All Areas</div></div>';
+			$(map.getDiv()).append(contextmenuDir);
+
+			setMenuXY(currentLatLng);
+
+			contextmenuDir.style.visibility = "visible";
+		}
+		function showContextMenu2(currentLatLng,currentId) {
+			var projection;
+			var contextmenuDir;
+			projection = map.getProjection() ;
+			$('.cmenu').remove();
+			contextmenuDir = document.createElement("div");
+			contextmenuDir.className  = 'cmenu';
+			contextmenuDir.innerHTML = '<div class="item" title="Remove selected Area" onclick="removeArea(' + currentId + ');"><div class="label">Remove Area</div></div><div class="item ' + (locations.length > 0 ? '' : 'disabled') + '" title="Copy locations list to clipboard" ' + (locations.length > 0 ? 'onclick="copyWorkerLocations()"' : '') + '><div class="label">Copy Marker Locations</div></div><div class="item ' + (locations.length > 0 ? '' : 'disabled') + '" title="Clear all areas from the map" ' + (locations.length > 0 ? 'onclick="clearAllAreas()"' : '') + '><div class="label">Clear All Areas</div></div>';
+			$(map.getDiv()).append(contextmenuDir);
+
+			setMenuXY(currentLatLng);
+
+			contextmenuDir.style.visibility = "visible";
+		}
+		function hideContextMenu() {
+			$('.cmenu').remove();
+			$("#LocationsJSON").fadeOut(500);
+		}
+		function addNewArea(Lat,Lng) {
+			var rectangle = new google.maps.Rectangle({
+				bounds: {
+					north: Lat,
+					south: Lat - 0.008,
+					east: Lng + 0.01,
+					west: Lng
+				},
+				map: map,
+				editable: true
```

Comment:
So, I tried that...but when I allowed draggable it broke the delete function for some reason. 

When the file was first open it would work perfectly, but after deleting an area none of the remaining areas would become non-draggable. 

BTW, your work on the ConfigGenerator over at [/bunzosteele/PokeWatch](https://github.com/bunzosteele/PokeWatch) is impressive. I'm a complete novice at this, so if you can figure out what was causing the bug, I would love to make them draggable.